src/core/ucc_team.c

```diff
@@ -229,3 +237,114 @@ ucc_status_t ucc_team_destroy(ucc_team_h team)
     }
     return status;
 }
+
+ucc_status_t ucc_ee_create(ucc_team_h team, const ucc_ee_params_t *params,
+                           ucc_ee_h *ee_p)
+{
+    ucc_ee_t *ee;
+
+    ee = ucc_malloc(sizeof(ucc_ee_t), "ucc execution engine");
+    if (!ee) {
+        ucc_error("failed to allocate %zd bytes for ucc execution engine",
+                  sizeof(ucc_ee_t));
+        return UCC_ERR_NO_MEMORY;
+    }
+
+    ee->team = team;
+    ee->ee_type = params->ee_type;
+    ee->ee_context_size = params->ee_context_size;
+    ee->ee_context = params->ee_context;
+    ucc_spinlock_init(&ee->lock, 0);
+    ucc_queue_head_init(&ee->event_in_queue);
+    ucc_queue_head_init(&ee->event_out_queue);
+    *ee_p = ee;
+
+    ucc_info("ucc_ee_create: ee is created ");
+
+    return UCC_OK;
+}
+
+ucc_status_t ucc_ee_destroy(ucc_ee_h ee)
+{
+    ucc_spinlock_destroy(&ee->lock);
+    ucc_free(ee);
+
+    ucc_info("ucc_ee_create: ee is destroyed ");
+
+    return UCC_OK;
+}
+
+ucc_status_t ucc_ee_get_event_internal(ucc_ee_h ee, ucc_ev_t **ev, ucc_queue_head_t *queue)
+{
+    ucc_event_desc_t *event_desc;
+    ucc_queue_elem_t *elem;
+
+    ucc_spin_lock(&ee->lock);
+    if (ucc_queue_is_empty(queue)) {
+        ucc_spin_unlock(&ee->lock);
+        return UCC_ERR_NOT_FOUND;
+    }
+
+    elem = ucc_queue_pull(queue);
+    ucc_spin_unlock(&ee->lock);
+
+    ucc_assert(elem);
+
+    event_desc = ucc_container_of(elem, ucc_event_desc_t, queue);
+    *ev = &event_desc->ev;
+
+    ucc_info("EE Event Get. ee:%p, queue:%p ev_type:%s ",
+                ee, queue, ucc_ee_ev_names[event_desc->ev.ev_type]);
+    return UCC_OK;
+}
+
+ucc_status_t ucc_ee_get_event(ucc_ee_h ee, ucc_ev_t **ev)
```

Comment:
i think all these functions should go to core/ucc_ee.c, not into team.c

Reply:
fixed